test/e2e/e2e_test.go

```diff
@@ -323,6 +324,89 @@ func testE2E() {
 		}).Should(Succeed())
 	})
 
+	It("should react to failure devices", func(ctx SpecContext) {
+		storageClass := "topolvm-provisioner-volumehealth"
+		By(fmt.Sprintf("deploying Pod with PVC based on StorageClass: %s", storageClass))
+		claimYAML := []byte(fmt.Sprintf(pvcTemplateYAML, "topo-pvc", "Filesystem", 200, storageClass))
+		podYaml := []byte(fmt.Sprintf(podVolumeMountTemplateYAML, "ubuntu", "topo-pvc"))
+
+		_, err := kubectlWithInput(claimYAML, "apply", "-n", ns, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred())
+		_, err = kubectlWithInput(podYaml, "apply", "-n", ns, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		By("confirming that the lv correspond to LogicalVolume resource is registered in LVM")
+		Eventually(func() error {
+			var pvc corev1.PersistentVolumeClaim
+			err = getObjects(&pvc, "pvc", "-n", ns, "topo-pvc")
+			Expect(err).ShouldNot(HaveOccurred())
+			return checkLVIsRegisteredInLVM(pvc.Spec.VolumeName)
+		}).Should(Succeed())
+
+		By("triggering a volume health partial activation failure")
+		var failureDeviceCount int
+		if nonControlPlaneNodeCount == 0 {
+			failureDeviceCount = 1
+		} else {
+			failureDeviceCount = nonControlPlaneNodeCount
+		}
+		for i := 0; i < failureDeviceCount; i++ {
+			out, err := exec.CommandContext(
+				ctx,
+				"sudo",
+				"dmsetup",
+				"remove",
+				"-f",
+				// This is the device name that is used in the volume health test vg for the crypt setup
+				fmt.Sprintf("/dev/mapper/crypt-%v", i+1),
+			).CombinedOutput()
+			if err != nil {
+				GinkgoT().Logf(err.Error())
```

Comment:
This only logs the error message and the test code continues. It would be better to assert this, IMO.
The error from execAtLocal contains the original error and stdout/stderr, it will show us sufficient information if an assertion will fail.